src/LtiAssignmentsGradesService.php

```diff
@@ -51,19 +51,34 @@ public function findOrCreateLineitem(LtiLineitem $new_line_item)
         if (!in_array(LtiConstants::AGS_SCOPE_LINEITEM, $this->service_data['scope'])) {
             throw new LtiException('Missing required scope', 1);
         }
-        $line_items = $this->service_connector->makeServiceRequest(
-            $this->service_data['scope'],
-            LtiServiceConnector::METHOD_GET,
-            $this->service_data['lineitems'],
-            null,
-            null,
-            'application/vnd.ims.lis.v2.lineitemcontainer+json'
-        );
+        $line_items = [];
+
+        $next_page = $this->service_data['lineitems'];
+
+        while ($next_page) {
+            $page = $this->service_connector->makeServiceRequest(
+                $this->service_data['scope'],
+                LtiServiceConnector::METHOD_GET,
+                $next_page,
+                null,
+                null,
+                'application/vnd.ims.lti-gs.v1.contextgroupcontainer+json'
+            );
+
+            $line_items = array_merge($line_items, $page['body']);
+            $next_page = false;
+            $link = $page['headers']['Link'];
+            if (preg_match(LtiServiceConnector::NEXT_PAGE_REGEX, $link, $matches)) {
+                $next_page = $matches[1];
+            }
+        }
+
```

Comment:
Oh... also. I assume all of this will get replaced with a call to `$this->getLineItems();`?

Reply:
.........

uhh yep totally planned on doing that :eyes: :grin: